alf/environments/suite_socialbot.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2019 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+try:
+    import social_bot
+except ImportError:
+    social_bot = None
+
+import contextlib
+import socket
+import gym
+from fasteners.process_lock import InterProcessLock
+from tf_agents.environments import suite_gym, wrappers, parallel_py_environment
+import gin.tf
+
+DEFAULT_SOCIALBOT_PORT = 11345
+
+
+def is_available():
+    return social_bot is not None
+
+
+@gin.configurable
+def load(environment_name,
+         port=None,
+         wrap_with_process=True,
+         discount=1.0,
+         max_episode_steps=None,
+         gym_env_wrappers=(),
+         env_wrappers=(),
+         spec_dtype_map=None):
+    """Loads the selected environment and wraps it with the specified wrappers.
+
+    Note that by default a TimeLimit wrapper is used to limit episode lengths
+    to the default benchmarks defined by the registered environments.
+
+    Args:
+        environment_name: Name for the environment to load.
+        port: Port used for the environment
+        wrap_with_process: Whether wrap environment in a new process
+        discount: Discount to use for the environment.
+        max_episode_steps: If None the max_episode_steps will be set to the default
+            step limit defined in the environment's spec. No limit is applied if set
+            to 0 or if there is no timestep_limit set in the environment's spec.
+        gym_env_wrappers: Iterable with references to wrapper classes to use
+            directly on the gym environment.
+        env_wrappers: Iterable with references to wrapper classes to use on the
+            gym_wrapped environment.
+        spec_dtype_map: A dict that maps gym specs to tf dtypes to use as the
+            default dtype for the tensors. An easy way how to configure a custom
+            mapping through Gin is to define a gin-configurable function that returns
+            desired mapping and call it in your Gin config file, for example:
+            `suite_socialbot.load.spec_dtype_map = @get_custom_mapping()`.
+
+    Returns:
+        A PyEnvironmentBase instance.
+    """
+
+    gym_spec = gym.spec(environment_name)
+    if max_episode_steps is None and gym_spec.timestep_limit is not None:
+        max_episode_steps = gym_spec.max_episode_steps
```

Comment:
It dose not work because of [these lines ](https://github.com/tensorflow/agents/blob/0d4ef352235a6961255f17262f602de798f9c3ae/tf_agents/environments/suite_gym.py#L126-L127) in `tf_agents.environments.suite_gym.wrap_env`

set `max_episode_steps` to 0 when it's None fix the problem